### PR TITLE
Spark Submit changes and test

### PR DESCRIPTION
**Pull request by foxish**
cc @mccheah @liyinan926 @apache-spark-on-k8s/contributors

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala [hunk 1]
```diff
@@ -381,6 +381,29 @@ class SparkSubmitSuite
...
+    val appArgs = new SparkSubmitArguments(clArgs)
+    val (childArgs, classpath, sysProps, mainClass) = prepareSubmitEnvironment(appArgs)
+    val childArgsMap = childArgs.grouped(2).map(a => a(0) -> a(1)).toMap
+    childArgsMap.contains("--primary-java-resource") should be (true)
+    childArgsMap.get("--main-class") should be (Some("org.SomeClass"))
+    childArgsMap.get("--arg") should be (Some("arg1"))
+
+    mainClass should be ("org.apache.spark.deploy.k8s.submit.Client")
+    classpath should have length (0)
+    sysProps("spark.executor.memory") should be ("5g")
```
Comment (liyinan926) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala]: Add one for `spark.driver.memory`.

**Review by liyinan926 — COMMENTED** [review]
LGTM with one minor comment.

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala (hunk 1 above)
Comment (foxish) [reply core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala]: Done. Thanks!

**Comment by ifilonenko** [comment]
Are we missing testing logic for app resources or is that going to be a follow up PR?

**Comment by foxish** [comment]
@ifilonenko Added the check for the jar that's submitted. Did you mean something else? The implementation details wrt resources should be covered under `org/apache/spark/deploy/k8s/submit/ClientSuite.scala` and submission step tests I think.

**Comment by ifilonenko** [comment]
@foxish that was part of it, yes. But coverage in regards to this: https://github.com/apache-spark-on-k8s/spark/blob/0add46a0240d74fb4f746e100ee9a7d6d29d87bd/core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala#L636-L653

**Comment by foxish** [comment]
Ah! I see.. I think those should be separate unit tests in a subsequent PR. 
I was looking at this from the perspective of the upstreaming, and initially, we're not going to have python/R support.

**Review by mccheah — APPROVED** [review]

**Comment by foxish** [comment]
rerun integration test please

**Comment by foxish** [comment]
rerun unit tests please

**Comment by foxish** [comment]
rerun integration test please

**Comment by foxish** [comment]
rerun integration test please

**Pull request merged**